File: Interviews/LeetCode/Topic-Wise/STL/Easy/Contains-Duplicate-II/219.cpp
class Solution {
public:
    bool containsNearbyDuplicate(vector<int>& nums, int k) {
        unordered_map<int, int> lastIndex;
        for (int i = 0; i < nums.size(); i++) {
            if (lastIndex.find(nums[i]) != lastIndex.end()) {
                if (i - lastIndex[nums[i]] <= k) return true;
            }
            lastIndex[nums[i]] = i;
        }

        return false;
    }
};

File: Interviews/LeetCode/Daily-Problems/2022/October/Day-21/219.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        lastIndex = {}
        for i in range(len(nums)):
            if nums[i] in lastIndex:
                if i - lastIndex[nums[i]] <= k: return True
            lastIndex[nums[i]] = i
        
        return False